examples/player/simid_player.js

```diff
@@ -609,14 +592,68 @@ class SimidPlayer {
     console.log('The creative has asked for the player to ping ' + requestedUrlArray);
   }
 
+  /**
+   * Called when video playback is complete.
+   * @private
+   */
+  videoComplete() {
+    this.simidProtocol.sendMessage(MediaMessage.ENDED);
+
+    if (this.requestedDuration_ == NO_REQUESTED_DURATION) {
+      this.stopAd(StopCode.MEDIA_PLAYBACK_COMPLETE);
+    }
+  }
+
+  /**
+   * Called when creative requests a change in duration of ad.
+   * @private
+   */
   onRequestChangeAdDuration(incomingMessage) {
-    if (this.requestedDuration_ != NO_REQUESTED_DURATION) {
-      // TODO: Support multiple change duration requests.
-      this.simidProtocol.reject(incomingMessage);
+    this.requestedDuration_  = incomingMessage.args['duration'];
+    //If requested duration is negative seconds, play entire ad
+    if (this.requestedDuration_ != UNLIMITED_DURATION && this.requestedDuration_< 0) {
+      this.sendLog("requested duration too short: " + this.requestedDuration_ + " seconds");
+      setTimeout(() => {
+        this.stopAd(StopCode.MEDIA_PLAYBACK_COMPLETE);
+        this.simidProtocol.reject(incomingMessage);
+      }, this.adVideoElement_.duration * 1000);
+    }//If requested duration is positive amount of seconds, call changeAdEndTime()
+    else {
+      this.changeAdEndTime();
+      this.simidProtocol.resolve(incomingMessage);
     }
-    const requestedDuration  = incomingMessage.args['duration'];
-    this.requestedDuration_ = requestedDuration;
-    this.simidProtocol.resolve(incomingMessage);
+  }
+
+  /**
+   * Compares the requested change duration with the current ad time
+   *  to determine when to stop the ad, every 250 milliseconds.
+   * @private
+   */
+  changeAdEndTime() {
+    this.interval_ = setInterval(() => {
+        this.compareAdAndRequestedDurations(); 
+    }, INTERVAL_TIMER);
+  }
+
+  /**
+   * Compares the duration of the ad with the requested change duration.
+   * If request duration is shorter, the ad stops early. 
+   * If the request duration is longer, the ad extends for the requested
+   *  amount of time.
+   * @private
+   */
+  compareAdAndRequestedDurations() {
+    if (this.adVideoElement_.currentTime >= this.requestedDuration_) {
+        this.stopAd(StopCode.CREATIVE_INITATED);
+        clearInterval(this.interval_);
+    } else if (this.requestedDuration_ >= this.adVideoElement_.duration) {
+      const durationChangeMs = (this.requestedDuration_ - this.adVideoElement_.duration) * 1000;
+      setTimeout(() => {
+        this.stopAd(StopCode.CREATIVE_INITIATED);
+      }, durationChangeMs);
+      clearInterval(this.interval_);
+    }
+    this.simidProtocol.sendMessage(MediaMessage.ENDED);
```

Comment:
we can remove this because now we're sending it directly when we get the ended message from the video element.

Reply:
okay